msg.go

```diff
@@ -921,7 +921,7 @@ func packStructValue(val reflect.Value, msg []byte, off int, compression map[str
 				copy(msg[off:off+len(s)], s)
 				off += len(s)
 			case `dns:"octet"`:
-				bytesTmp := make([]byte, 0)
+				bytesTmp := make([]byte, 256*4+1)
```

Comment:
I actually meant to add a comment asking about this, I just borrowed the buffer size here from the size used for TXT record contents. CAA value bodies will *generally* contain a domain name, but can contain arbitrary data, from a quick scan at the RFC they don't mention a strict limit just saying

```
The length of the value field is specified implicitly as the remaining length of the enclosing Resource Record data field.
``` 

It might be best to set the buffer size to 255, which I believe is the maximum size for a single string in a TXT or SPF record, since from at least what I've seen the value should generally take up far less.